src/Channels.Networking.Sockets/Signal.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Channels.Networking.Sockets
```

Comment:
@davidfowl  I wanted to talk about that; this has clear exact-same-scenario usage in the Rio code; we don't have a `Channels.Networking` assembly, but I planned to submit a second PR to move this, promoting it to the `Channels.Networking` namespace in the `Channels` assembly. Obviously this would then be publicly exposed, hence wanted to discuss it with you more.

If it is OK with you, I'd rather keep it on here to keep the code compiling, so we can close down this PR that is getting over-long already - while we discuss that separately.